Providing access to the translator so that custom rule objects can be…

jcrawford opened the pull request:
… used and translated

Commit:
Providing access to the translator so that custom rule objects can be used and translated

jeffochoa commented on src/ValidatorFactory.php:
```diff
@@ -53,11 +55,26 @@ public function loadTranslator() : Translator
+     * @param string $key
+     * @return string
+     */
+    public static function trans(string $key): string
```
I don't think this `trans()` function belongs to the current class 🤔

Actually, I don't believe that it is necessary because now with the implementation above you could just do `$validator->translator->trans($key)`, right?

jcrawford replied:
Yes you could do that but if you ever change the property to protected and not public then it would break.  I typically only expose functionality that my classes provide rather than making the dependencies public but if you wish I can make it work the way you wish.

The code in the rule objects would have to be the following.

```php
ValidatorFactory::$translator->trans($key);
```

jcrawford replied:
I will add some tests in the morning but it will require some code such as a custom rule object to be created.

jeffochoa reviewed (commented):
Thank you so much 🙌 

Would you like to add some tests? 🙂

see https://github.com/jeffochoa/validator-factory/blob/master/tests/Unit/ValidatorTest.php

Commit:
Adding tests for the custom rules

Commit:
Adding a test to check that the key is returned on failed translation

Commit:
Removing the trans method from the Validator Factory and requiring Rule Objects to call ValidatorFactory::->trans() instead

jcrawford commented:
I have added some tests and also refactored the code to use return ValidatorFactory::$translator->trans($key); rather than having a trans() method on the ValidatorFactory as you requested.
